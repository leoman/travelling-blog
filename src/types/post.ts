import moment from 'moment';
import { Location } from './location'
import { Photo } from './photo'

export enum Status {
  live = "live",
  draft = "draft"
}

export interface Post {
  id?: number
  title: string
  slug?: string
  titleColour?: string
  content?: string
  date?: Date | string
  order?: Date | string
  photo?: string
  status?: Status
  location?: Location
  photos?: Photo[]
}

export interface GeoPost extends Post {
  y: number
  x: number
}

export const initialState: Post = {
  title: '',
  titleColour: '',
  content: '',
  photo: '',
  date: new Date(),
  order: new Date(moment().format('YYYY-MM-DD')),
  status: Status.draft,
  location: { 
      location: '',
      duration: 0,
      lat: 0, 
      lng: 0,
      hideFromBounding: false,
  },
  photos: [],
}
