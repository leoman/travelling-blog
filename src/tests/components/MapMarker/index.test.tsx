import React from "react"
import { BrowserRouter as Router } from "react-router-dom"
import { mount } from 'enzyme'
import { act } from "react-dom/test-utils"
import renderer from 'react-test-renderer'
import '../../setup'
import MapMarker from '../../../components/Map/MapMarkers/MapMarker'
import { MapMarkerI } from '../../../types'

let wrapper = null
beforeEach(() => {
  wrapper = null
})

afterEach(() => {
  wrapper = null
})

const markerProps: MapMarkerI = {
  post: {
    title: 'Tongariro Crossing',
    slug: 'Tongariro-Crossing',
    photo: 'https://live.staticflickr.com/1959/44695030985_da4b577d9a_b.jpg',
  },
  lat: -21.209277,
  lng: -159.749581,
  hovered: true
}

describe('MapMarker', () => {

  it('should render when given the correct props', () => {
    const { post, hovered, lat, lng } = markerProps
    const tree = renderer
      .create(<Router><MapMarker post={post} hovered={hovered} lat={lat} lng={lng} /></Router>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should be highlighted when that is passed in true', () => {
    const { post, hovered, lat, lng } = markerProps
    act(() => {
      wrapper = mount(<Router><MapMarker post={post} hovered={hovered} lat={lat} lng={lng} /></Router>)
    })

    expect(wrapper.find('MapMarker').prop('hovered')).toEqual(true)
  })
})