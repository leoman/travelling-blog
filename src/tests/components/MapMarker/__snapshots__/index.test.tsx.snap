// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapMarker should render when given the correct props 1`] = `
<div
  className="sc-AxjAm kruEfP"
>
  <a
    className="sc-AxirZ dWhLQ"
    hovered="true"
    href="/posts/Tongariro-Crossing"
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(https://live.staticflickr.com/1959/44695030985_da4b577d9a_b.jpg)",
      }
    }
  />
</div>
`;
